src/test/java/org/testeditor/fixture/commons/text/generate/NameGeneratorTest.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2018 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+
+package org.testeditor.fixture.commons.text.generate;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+class NameGeneratorTest {
+
+    @Test
+    void generateRandomFirstNameTest() {
+        String randomName = NameGenerator.getRandomFirstName();
+        Assert.assertNotNull(randomName);
+        Assert.assertTrue(randomName.length() > 0);
+    }
+    
+    @Test
+    void generateFirstNameWithSeedTest() {
+        String randomName = NameGenerator.getFirstName(32);
+        Assert.assertEquals("Ugochi", randomName);
+    }
+      
+    @Test
+    void generateRandomFullname() {
+        String randomName = NameGenerator.getRandomFullName();
+        Assert.assertNotNull(randomName);
+        Assert.assertTrue(randomName.length() > 0);
+        //\p{L} - any Unicode letter. \p{M} - any diacritic. \s - whitespace, ' and - are literal single quote and -
+        Assert.assertTrue(randomName.matches("(?U)[\\p{L}\\p{M}]+(?:[\\s'-][\\p{L}\\p{M}]+)*"));
+    }
+
+    @Test
+    void generateFullnameWithSeed() {
+        String randomName = NameGenerator.getFullName(32);
+        Assert.assertEquals("Ugochi Strohmeier", randomName);
+    }
+    
+    @Test
+    void generateRandomLastNameTest() {
+        String randomName = NameGenerator.getRandomLastName();
+        Assert.assertNotNull(randomName);
+        Assert.assertTrue(randomName.length() > 0);
+    }
+    
+    @Test
+    void generateLastNameWithSeedTest() {
+        String randomName = NameGenerator.getLastName(32);
+        Assert.assertEquals("Strohmeier", randomName);
+    }
+    
+    @Test
+    void generateRandomCompanyNameTest() {
+        String randomName = NameGenerator.getRandomCompanyName();
+        Assert.assertNotNull(randomName);
+        Assert.assertTrue(randomName.length() > 0);
+    }
+    
+    @Test
+    void generateCompanyNameWithSeedTest() {
+        String randomName = NameGenerator.getCompanyName(32);
+        Assert.assertEquals("Fortune Brands Home & Security", randomName);
+    }
+    
+    @Test
+    void generateRandomPersonFieldsNameTest() {
+        String[] personFields = NameGenerator.getRandomPersonFields();
+        Assert.assertNotNull(personFields);
+        Assert.assertTrue(personFields.length == 5);
```

Comment:
~So, maybe the method should be called `getFiveRandomNames` or something…?~
Edit: ok, now I get what that method does. Still, not really obvious from the name